lib/system/nimscript.nim

```diff
@@ -58,14 +58,6 @@ proc rawExec(cmd: string): int {.tags: [ExecIOEffect], raises: [OSError].} =
 proc warningImpl(arg, orig: string) = discard
 proc hintImpl(arg, orig: string) = discard
 
```

Comment:
```suggestion
when (NimMajor, NimMinor) < (1, 3) or defined(nimdoc):
  from os import paramStr, paramCount
  export paramStr, paramCount
```

Reply:
@hlaaftana this PR is kinda abandoned because no unanimous decision could be formed as to what should be done regarding this, because 
1) every change was a breaking change
2) there are plenty of similar special procs for nimscript and just fixing these 2 was arbitrary, and cleaning up everything was a very major breaking change
  
So, there were discussions on irc/gitter regarding this, which varied from let's do the change for only these 2 procs, let's do full cleanup (full cleanup was not possible due to some technical reasons IIRC), the patch is good as-is, the patch was better before the force push, let's deprecate nimscript as a whole and use nim for configuring projects (à la Zig programming language and many others, but simpler because of macros in Nim).
But no decision was ever made.

cc @Araq, has any decision been made regarding this now?

Reply:
I agree with the idea of "borrowed procs should be imported from the standard library modules under `when defined(nimscript)`", and I agree that the other procs in this module should be moved too, but only the obvious offenders i.e. existsFile/Dir, all the Env procs, findExe, even getCurrentDir, so as to reduce bloat and ease confusion.

So:

* let's do the change for only these 2 procs - I say no, but shouldn't be that hard.
* let's do full cleanup - depends on what "full" means. there's no rush, simple changes are fine
* the patch is good as is - it's really easy to extend its reach
* the patch was better before the force push - disagree, `os` procs being in `os` and not somewhere else makes sense. not too ambitious at all
* let's deprecate nimscript as a whole - Until we can replace the VM with something as powerful as something like an LLVM JIT I think not. Though sometimes `nim js -r` ends up being way faster than `nim e`.

Come to think of it, I actually don't think there has to be a since annotation here, the old builtins should be removed and the code should just be replaced with

```nim
from os import getCurrentDir, paramStr, paramCount, getEnv, existsEnv, putEnv, delEnv, fileExists, dirExists, existsFile, existsDir, findExe
export getCurrentDir, paramStr, paramCount, getEnv, existsEnv, putEnv, delEnv, fileExists, dirExists, existsFile, existsDir, findExe
```

You can even go fancy, rename the nimscript magic procs from things like `rmDir` to `removeDir`, accordingly specialize `os.removeDir` for nimscript, then `import removeDir as rmDir`. `cd` isn't a builtin for example, it uses a proc from os.

Reply:
Cleaning up just these two procs is very good as long as it doesn't cause regressions.